### ISSUE TITLE
CTools in HBD for Avisynth+ and maybe VS

### ISSUE THREAD

**Issue by jonashed** [CeppoTools/CTools#2]
CTools is written for 8 bit depth only and the author has not been online in doom9 forum for almost one year.
I hereby request a rewrite/fork of CTools for HBD (up to at least YUV444 16 bits).
About CDegrain, this is the authors own words:
"This filter is a degrain that blends stationary parts of the picture.
It takes the chroma difference into account to reduce blending by using a block based approch."

Also I have a request for the CDegrain filter in the Ctools package, if it is possible to add a 'sClip' to this filter. My thoughts are as follows: CDegrain makes a temporal denoise/degrain only on static/non moving parts of the image but do not touch moving parts. My idea is to use an external clip like this example:

prefilt=fft3dfilter(sigma=3) smooth=blur(0.3) CDegrain(3,5,32,4,dClip=prefilt,sClip=smooth)

to filter only the parts that CDegrain do not touch (the moving parts) with the filter of your choice.

Thanks!

**Comment by kedaitinh12** [comment]
Here modify CDegrain
https://forum.doom9.org/showthread.php?p=1981310#post1981310

**Comment by CeppoTools** [comment]
I answered in Asd-g issue that you opened :)

**Comment by jonashed** [comment]
Thanks man great to hear it :D
While at it, is CDegrain good for real film (non anime)? I get good
denoising in some areas in my high ISO noisy DSLR footage but it misses
whole areas of noise. Probably need a stronger prefilter for dclip?

Thanks again!

Den mån 23 jan. 2023 22:31Ceppo ***@***.***> skrev:

**Comment by CeppoTools** [comment]
Nope, is for anime, unless you have the tools to make a static recording for the background, it is not going to work. I only play with anime, I actually don't encode nothing, I just like to have fun. You could try MCompensate with a low thsad to make static the background?

**Comment by jonashed** [comment]
Thanks yes I will try that :)

Den mån 23 jan. 2023 22:54Ceppo ***@***.***> skrev:

**Comment by CeppoTools** [comment]
@jonashed Did you try? How did it go?

**Comment by jonashed** [comment]
I will try tomorrow because Im on a job project until then :)
Would be cool to have a parameter to tweak a threshold of how static it
will denoise. Because now it is locked in totally static only? Hmm when
denoising anime how is temporal denoising working I wonder. Does not
usually the frames differ too much since the big gap with lower frame rates
and no motionblur? The backgrounds I gues can scroll.


Den ons 25 jan. 2023 17:36Ceppo ***@***.***> skrev:
